src/state-calculator.tsx

```diff
@@ -429,16 +304,160 @@ export const StateCalculator: FC<{
           />
         </>
       )}
-    </>
+    </div>
   );
 };
 
+class CalculatorElement extends HTMLElement {
+  /* supported property to switch UI language */
+  override lang: string = this.getDefaultLanguage();
+
+  /* property to show the email signup field */
+  showEmail: boolean = false;
+
+  /* property to include incentives from states that aren't formally launched */
+  includeBetaStates: boolean = false;
+
+  /* supported properties to control which API path and key is used to load the calculator results */
+  apiKey: string = '';
+  apiHost: string = DEFAULT_CALCULATOR_API_HOST;
+
+  /* supported property to allow restricting the calculator to ZIPs in a specific state */
+  state: string = '';
+
+  /* supported properties to allow pre-filling the form */
+  zip: string = DEFAULT_ZIP;
+  ownerStatus: OwnerStatus = DEFAULT_OWNER_STATUS;
+  householdIncome: string = DEFAULT_HOUSEHOLD_INCOME;
+  taxFiling: FilingStatus = DEFAULT_TAX_FILING;
+  householdSize: string = DEFAULT_HOUSEHOLD_SIZE;
+
+  /* attributeChangedCallback() will be called when any of these changes */
+  static observedAttributes = [
+    'lang',
+    'show-email',
+    'include-beta-states',
+    'api-key',
+    'api-host',
+    'state',
+    'zip',
+    'owner-status',
+    'household-income',
+    'tax-filing',
+    'household-size',
+  ] as const;
+
+  reactRootCalculator: Root;
+  reactRootFooter: Root;
+
+  constructor() {
+    super();
+
+    const shadowRoot = this.attachShadow({ mode: 'open' });
+
+    const style = document.createElement('style');
+    style.textContent = tailwindStyles;
```

Comment:
The old code includes `baseVariables` in the shadow styles, but that isn't necessary anymore.